beater/handlers.go

```diff
@@ -58,9 +58,12 @@ func backendHandler(p processor.Processor, config Config, report reporter) http.
 }
 
 func frontendHandler(p processor.Processor, config Config, report reporter) http.Handler {
-	return logHandler(
-		frontendSwitchHandler(config.EnableFrontend,
-			processRequestHandler(p, config, report)))
+	if config.EnableFrontend {
```

Comment:
im not convinced about this. 

i wanted to not have any logic at all on the top level handlers (backend, frontend) and just chain downstream handlers, so they can be better reused. chaining would be their only responsibility.
this ways you are doing too many things imo: composing _and_ branching.

i'd liked more eg. @simitt idea of renaming the feSwitchHandler so it could be used in the backend if needed. this way you would have to duplicate the logic, and is not like this affords  code or something.

Reply:
To me it's cleaner that the logic related to handling frontend endpoints is only in one place. The rest of the chain is generic.